File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        Morse = [".-","-...","-.-.","-..",".","..-.","--.",
                 "....","..",".---","-.-",".-..","--","-.",
                 "---",".--.","--.-",".-.","...","-","..-",
                 "...-",".--","-..-","-.--","--.."]
        
        seen = {"".join(Morse[ord(c) - ord('a')] for c in word) for word in words}
        return len(seen)
        

File: 804-unique-morse-code-words/804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] Morse =  new String[]{".-","-...","-.-.","-..",".","..-.","--.",
                         "....","..",".---","-.-",".-..","--","-.",
                         "---",".--.","--.-",".-.","...","-","..-",
                         "...-",".--","-..-","-.--","--.."};
                                       
        Set<String> seen = new HashSet();
        for (String word: words){
            StringBuilder code = new StringBuilder();
            for (char c: word.toCharArray())
                code.append(Morse[c - 'a']);
            seen.add(code.toString());
        }
        return seen.size();
    }
}